kotlintest-runner/kotlintest-runner-jvm/src/main/kotlin/io/kotlintest/runner/jvm/TestCaseExecutor.kt

```diff
@@ -41,9 +42,16 @@ class TestCaseExecutor(private val listener: TestEngineListener,
     try {
 
       // invoke the "before" callbacks here on the main executor
-      context.launch(executor.asCoroutineDispatcher()) {
-        before(testCase)
-      }.join()
+      try {
```

Comment:
Does this actually make any difference? Wouldn't an error inside before() just get silently dropped like before because the exception doesn't propagate out of the thread it runs on ?

Reply:
It does. JUnit is not displaying the test because it was never started.  I'm capturing the `before` error to start the test, and then I'll let it propagate further up the stack.

I don't want to start it if it didn't errored in `before`, as the test might be ignored. But if `before` is an error, it's not possible for it to be ignored.

The exception will propagate, as we'll rethrow it. We're just capturing it here to ensure the test is started (invoking)